src/docfx/lib/git/GitUtility.cs

```diff
@@ -121,6 +121,19 @@ public static Task Fetch(string path, string url, string committish, Config conf
             return CloneOrUpdate(path, url, new[] { committish }, bare: false, depthOne: false, prune: false, config);
         }
 
+        /// <summary>
+        /// Get a list of commits using git log
+        /// </summary>
+        public static Task<string[]> GetCommits(string path, int top = 1)
```

Comment:
this may take too long (consider azure-docs-pr with xxxxxxxxx of commits). consider use commit time for comparison. time could be stored in `DependencyGit` so you only get the time once on checkout.

Reply:
i only take top 100 for comparing

Reply:
> i only take top 100 for comparing

then it may give you wrong result

Reply:
commit time is not pushing time, not really correct.

Reply:
tree-depth?

Reply:
> > i only take top 100 for comparing
> 
> then it may give you wrong result

I believe it's enough, i then sort by last access time. considering our build frequency.

Reply:
considering perf, even i changed it to top 1000, it finished very fast, and it only happens on restoring per repo, not building per file.